src/main/java/com/mitchej123/hodgepodge/asm/transformers/mc/EntityPlayerTransformer.java

```diff
@@ -0,0 +1,105 @@
+package com.mitchej123.hodgepodge.asm.transformers.mc;
+
+import net.minecraft.launchwrapper.IClassTransformer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.FieldInsnNode;
+import org.objectweb.asm.tree.JumpInsnNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+
+public class EntityPlayerTransformer implements IClassTransformer {
+
+    private static final Logger LOGGER = LogManager.getLogger("EntityPlayerTransformer");
+
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass) {
+        if (basicClass == null) return null;
+
+        if (NameContext.Obf.entityPlayer.equals(name)) {
+            final ClassReader cr = new ClassReader(basicClass);
+            final ClassNode cn = new ClassNode();
+            cr.accept(cn, 0);
+            transformClassNode(NameContext.Obf, cn);
+            ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_MAXS);
+            cn.accept(cw);
+
+            LOGGER.info("Transformed {}", name);
+            return cw.toByteArray();
+        }
+
+        if (NameContext.Deobf.entityPlayer.equals(name)) {
+            final ClassReader cr = new ClassReader(basicClass);
+            final ClassNode cn = new ClassNode();
+            cr.accept(cn, 0);
+            transformClassNode(NameContext.Deobf, cn);
```

Comment:
I wouldn't do that to check the obfuscation state, check it from the injectData method in the `IFMLLoadingPlugin` implementation